dbms/src/Core/SpillHandler.cpp

```diff
@@ -96,6 +96,12 @@ void SpillHandler::spillBlocks(Blocks && blocks)
         {
             if (unlikely(!block || block.rows() == 0))
                 continue;
+            /// erase constant column
+            for (auto it = spiller->const_column_indexes.rbegin(); it != spiller->const_column_indexes.rend(); ++it) // NOLINT
```

Comment:
why not using `foreach`?
```
for (auto index : const_column_indexes)
{
    RUNTIME_CHECK_MSG(block.getByPosition(index).column->isColumnConst(), "The {}-th column in block must be constant column", *it);
    block.erase(index);
}
```

Reply:
Why is reverse order traversal required here?

Reply:
Because the index stored in `const_column_indexes` is based on the original `Block`, if the column before index is removed, the index need to be changed or it becomes invalid index.

Reply:
ok, maybe add some comment here?

Reply:
Done